File: backend/catalog-service/app/services/seat_service.py
from typing import List, Optional

from app.models import Seat
from app.repositories import SeatRepository


class SeatService:
    def __init__(self, repository: SeatRepository):
        self.repository = repository

    async def get_seat_by_id(self, seat_id: int) -> Optional[Seat]:
        return await self.repository.get_by_id(seat_id)

    async def get_seats_by_screen(self, screen_id: int) -> List[Seat]:
        return await self.repository.get_by_screen_id(screen_id)

File: backend/catalog-service/src/main/java/com/bookmyshow/catalog/service/SeatService.java
package com.bookmyshow.catalog.service;

import com.bookmyshow.catalog.entity.Seat;
import com.bookmyshow.catalog.repository.SeatRepository;
import com.bookmyshow.common.exception.ResourceNotFoundException;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Service
@Transactional
@RequiredArgsConstructor
public class SeatService {

    private final SeatRepository seatRepository;

    public Seat getSeatById(Long id) {
        return seatRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Seat", "id", id));
    }

    public List<Seat> getSeatsByScreen(Long screenId) {
        return seatRepository.findByScreenIdOrderByRowLabelAscSeatNumberAsc(screenId);
    }
}
